Dockerfile.doc

```diff
@@ -0,0 +1,30 @@
+FROM ubuntu:22.04
+
+# We use bash instead of /bin/sh as ESP-IDF doesn't support /bin/sh
+SHELL ["/bin/bash", "-c"]
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        apt-utils wget software-properties-common gnupg
+
+RUN env DEBIAN_FRONTEND=noninteractive \
+        apt-get update; \
+        apt-get install -y --no-install-recommends \
+            python3 python3-pip \
+            gcc g++ build-essential cmake make git graphviz flex bison libc6; \
+    git clone https://github.com/doxygen/doxygen.git /tmp/doxygen; \
+        cd /tmp/doxygen; \
+        git checkout 9b262a41ec2b1bcc2b64f7364730d164eeb3f2f5; \
+        mkdir build; \
+        cd build; \
+        cmake .. -DCMAKE_BUILD_TYPE=Release; \
+        make -j8; \
+        make install; \
+        rm -rf /tmp/doxygen \
+    apt-get remove gcc g++ build-essential
```

Comment:
Is there a reason to run `apt-get remove`?
Specifically confusing is the `remove build-essential`.

Reply:
To shrink the image. You no longer build C++ in the container, so you don't need them. A better solution would be to use build-stages by Dockerfile, but they weren't available on the target systems (now they are, but this is fairly simple Dockerfile).